### ISSUE TITLE
enterprise connectionpool - documentate run* - Methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7572]
|<img alt="malamute's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| malamute reported|
|-|-|
|Trac ID|trac#301|
|Type|enhancement|
|Created|2003-09-30 12:31:41Z|

Attachments:

* [twisted.enterprise.adbapi.py.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/787/787d41d9c35c57ef9e4aba799bacefac312149a4/f0580ff8a7fed8b21840942576eb35f15b3619ba.patch) (1817 bytes) - added by malamute on 2003-09-30 12:31:41Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__301 301
type__enhancement enhancement
reporter__malamute malamute
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1064925101000000 1064925101000000
changetime__1468272489897551 1468272489897551
version__ 
owner__ 
cc__davep cc__itamarst cc__malamute
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="malamute's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|malamute commented|
|-|-|

```
#!html
<pre>
I only copied the normal execute - part and changed it to 
executemany (also the doc-strings). I tested it on MySQLdb 
0.9.2 and it seems to work fine.

</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
runInteraction should cover this use case adequately, I
think: 
http://itamarst.org/writings/OSCON03/twisted_internet-102.html

If this is the case, please close the bug, or change the
title to "runInteraction needs better documentation" ;)

</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="malamute's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|malamute commented|
|-|-|

```
#!html
<pre>
Ok. runInteraction solves that problem quiet good. So this should really be
better documentaded. I will see if i can write a little documentation in the
next week about the run* - statments in ConnectionPool, because if i knew that
before i would have done many things diffrent. I think other people will go in
the same trap.

Thank you
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jdavisp3's avatar" src="https://avatars.githubusercontent.com/u/43582?s=50" width="50" height="50">](https://github.com/jdavisp3)|@jdavisp3 commented|
|-|-|

```
#!html
<pre>
Thank you, better documentation would be nice since
you're definitely not the first person to miss runInteraction
on the first pass.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
I added example of runInteraction.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
And the API docs are now generated correctly so you can see docs for run* methods.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix commented|
|-|-|

```
#!html
<pre>
shouldn't this issue be resolved?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jdavisp3's avatar" src="https://avatars.githubusercontent.com/u/43582?s=50" width="50" height="50">](https://github.com/jdavisp3)|@jdavisp3 commented|
|-|-|

```
#!html
<pre>
Yes, I think the new docs should cover this issue.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_10"></a>|GitHub commented|
|-|-|

In changeset ebcabbbba6bc899a9f8729a67a41a3f925a219b2
```
#!CommitTicketReference repository="" revision="ebcabbbba6bc899a9f8729a67a41a3f925a219b2"
Merge pull request #301 from twisted/4926-rodrigc-iocp-py3

Author: rodrigc
Reviewer: adiroiban
Fixes: #4926

Eliminate Windows iocp compilation errors on Python 3
```
